backend/internal/middleware/auth_test.go

```
@@ -0,0 +1,208 @@
+		}
+	}
+
+	// Test handler that checks if user is in context
+	testHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		user := r.Context().Value("user").(*model.User)
+		if user == nil {
+			http.Error(w, "user not found in context", http.StatusInternalServerError)
+			return
+		}
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("success"))
```

golangci-lint (1.64.8)

Error return value of `w.Write` is not checked

(errcheck)